globalconf.go

```diff
@@ -98,14 +98,14 @@ func (g *GlobalConf) Delete(flagSetName, flagName string) error {
 	return nil
 }
 
-// Parses the config file for the provided flag set.
+// ParseSet parses the config file for the provided flag set.
 // If the flags are already set, values are overwritten
 // by the values in the config file. Defaults are not set
 // if the flag is not in the file.
 func (g *GlobalConf) ParseSet(flagSetName string, set *flag.FlagSet) {
```

Comment:
Globally registering a flagset is not always what users can do. We should leave the decision to the user. We should maybe remove registration and ParseAll rather than removing the fundamental flagset-scoped ParseSet.